// @saeris/vue-spinners/example/dist/js/app.3001ca8e.js
(function(e){function t(t){for(var i,s,a=t[0],p=t[1],l=t[2],d=0,u=[];d<a.length;d++)s=a[d],o[s]&&u.push(o[s][0]),o[s]=0;for(i in p)Object.prototype.hasOwnProperty.call(p,i)&&(e[i]=p[i]);c&&c(t);while(u.length)u.shift()();return n.push.apply(n,l||[]),r()}function r(){for(var e,t=0;t<n.length;t++){for(var r=n[t],i=!0,a=1;a<r.length;a++){var p=r[a];0!==o[p]&&(i=!1)}i&&(n.splice(t--,1),e=s(s.s=r[0]))}return e}var i={},o={app:0},n=[];function s(t){if(i[t])return i[t].exports;var r=i[t]={i:t,l:!1,exports:{}};return e[t].call(r.exports,r,r.exports,s),r.l=!0,r.exports}s.m=e,s.c=i,s.d=function(e,t,r){s.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},s.r=function(e){"undefined"!==typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},s.t=function(e,t){if(1&t&&(e=s(e)),8&t)return e;if(4&t&&"object"===typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(s.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var i in e)s.d(r,i,function(t){return e[t]}.bind(null,i));return r},s.n=function(e){var t=e&&e.__esModule?function(){return e["default"]}:function(){return e};return s.d(t,"a",t),t},s.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},s.p="/";var a=window["webpackJsonp"]=window["webpackJsonp"]||[],p=a.push.bind(a);a.push=t,a=a.slice();for(var l=0;l<a.length;l++)t(a[l]);var c=p;n.push(["0dbb","chunk-vendors"]),r()})({"0dbb":function(e,t,r){"use strict";r.r(t);r("dbb5");var i,o=r("88d8"),n=r("9e7f"),s=r("8347"),a=r.n(s),p=r("821f"),l=r.n(p),c=r("3ba8");const d=Object(c["a"])(`section`)`
  margin: 50px auto;
  color: #fff;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  align-items: center;
  cursor: pointer;

  span {
    background: #2b303b;
    padding: 16px 32px;
    font-size: 20px;
    margin: 0 auto;
    font-family: Courier New, Courier, monospace;
    opacity: 0.9;
    transition: 0.3s ease-out;

    &:hover {
      box-shadow: 0 3px 3px 0 rgba(0, 0, 0, 0.14),
        0 1px 7px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -1px rgba(0, 0, 0, 0.2);
      opacity: 1;
    }
  }
`;let u=a()(i=class extends n["default"]{constructor(...e){var t;return t=super(...e),Object(o["a"])(this,"text",["yarn add @saeris/vue-spinners","npm install @saeris/vue-spinners --save"]),Object(o["a"])(this,"index",0),t}handleClick(e){e.preventDefault(),this.index=+!this.index}render(){const e=arguments[0];return e(d,[e("span",{on:{click:this.handleClick}},[this.text[this.index]])])}})||i;var h,x,b=r("8647"),f=r("6f66"),g=r.n(f);n["default"].use(g.a);const y=Object(c["a"])(`section`)`
  top: ${({pos:e})=>`${e}px`};
  left: 20px;
  position: fixed;
  z-index: 999;
`,m=Object(c["a"])(`button`)`
  height: 36px;
  font-size: 14px;
  line-height: 36px;
  background: #2b303b;
  color: #fff;
  border: none;
  opacity: 0.9;
  -webkit-transition: 0.5s ease-in-out;
  -o-transition: 0.5s ease-in-out;
  transition: 0.5s ease-in-out;
  font-weight: 300;
  text-transform: uppercase;
  cursor: pointer;
  border-radius: 2px;
  letter-spacing: 0.5px;
  padding: 0 28px;

  &:hover {
    box-shadow: 0 3px 3px 0 rgba(0, 0, 0, 0.14),
      0 1px 7px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -1px rgba(0, 0, 0, 0.2);
    opacity: 1;
  }
`;let w=(h=a()({props:{color:{type:String,required:!0},updateColor:{type:Function,required:!0}}}),h(x=class extends n["default"]{constructor(...e){var t;return t=super(...e),Object(o["a"])(this,"pos",370),Object(o["a"])(this,"showPicker",!1),t}beforeMount(){window.addEventListener("scroll",this.updatePosition,!0)}beforeDestroy(){window.removeEventListener("scorll",this.updatePosition,!0)}updatePosition(){let e=370-2*window.scrollY;this.pos=e>60?e:50}togglePicker(){this.showPicker=!this.showPicker}render(){const e=arguments[0];return e(y,{attrs:{pos:this.pos}},[this.showPicker?e(b["Sketch"],{directives:[{name:"click-outside",value:this.togglePicker}],attrs:{value:this.color},on:{input:this.updateColor}}):e(m,{on:{click:this.togglePicker}},["Change Color"])])}})||x);var v,j;const O=Object(c["a"])(`main`)`
  .spinners {
    margin-top: ${({isFixed:e})=>`${e?250:0}px`}
  }
`,k=Object(c["a"])(`header`)`
  position: realtive;
  display: flex;
  align-items: center;
  width: 100%;
  height: ${({height:e})=>`${e}px`};
  color: #fff;
  z-index: 1000;
  background: linear-gradient(90deg, ${({color:e})=>e}, #2b303b);

  &.fixed {
    position: fixed;
    top: 0;
    box-shadow: rgba(0, 0, 0, 0.14) 0px 3px 3px 0px,
      rgba(0, 0, 0, 0.12) 0px 1px 7px 0px, rgba(0, 0, 0, 0.2) 0px 3px 1px -1px;
  }
`,S=Object(c["a"])(`h1`)`
  height: 50px;
  line-height: 50px;
  display: block;
  text-align: center;
  margin: 0px auto;
  font-size: 40px;
  letter-spacing: 5px;
  font-weight: 300;
  text-transform: uppercase;
  position: -webkit-sticky;
  position: sticky;
  top: 10px;
  transition: 0.2s ease-in-out;

  &.fixed {
    font-size: 20px;
  }
`,P=Object(c["a"])(`footer`)`
  padding: 30px;
  margin: 15px 0 0 0;
  background: #2b303b;
  text-align: center;

  a {
    text-decoration: none;
    color: #ccc;
  }
`,F=Object(c["a"])(`img`)`
  position: absolute;
  top: 0;
  right: 0;
  border: 0;
`;let C=(v=a()({props:{color:{type:String,required:!0}}}),v(j=class extends n["default"]{constructor(...e){var t;return t=super(...e),Object(o["a"])(this,"isFixed",!1),Object(o["a"])(this,"height",360),t}beforeMount(){window.addEventListener("scroll",this.handleScroll,!0)}beforeDestroy(){window.removeEventListener("scorll",this.handleScroll,!0)}handleScroll(){let e=360-window.scrollY;e>200?(this.height=e,this.isFixed=!1):(this.height=40,this.isFixed=!0)}render(){const e=arguments[0];return e(O,{attrs:{isFixed:this.isFixed}},[e(k,{class:{fixed:this.isFixed},attrs:{height:this.height,color:this.color}},[this.isFixed?null:e("a",{attrs:{href:"https://github.com/Saeris/vue-spinners"}},[e(F,{attrs:{src:"https://s3.amazonaws.com/github/ribbons/forkme_right_green_007200.png",alt:"Fork me on GitHub"}})]),e(S,{class:{fixed:this.isFixed}},["Vue Spinners"])]),this.$slots.default,e(P,[e("a",{attrs:{href:"https://github.com/Saeris/vue-spinners"}},["GitHub Repository"])])])}})||j);var z=r("9395");const $=Object(c["a"])(`div`)`
  position: absolute;
  bottom: 5px;
  font-family: sans-serif, serif;
  width: 80%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
`,_=Object(c["a"])(`div`)`
  margin: 3px;
  display: flex;
  width: 50px;
  height: 20px;
  position: relative;
`,q=Object(c["a"])(`input`)`
  width: 50px;
  padding: 0 4px;
  display: block;
  border: none;
  border-bottom: 1px solid #757575;
  text-align: center;

  &:focus {
    outline: none;
  }
`,M=Object(c["a"])(`span`)`
  position: absolute;
  display: block;
  width: 100%;
  bottom: -1px;

  &::before {
    left: 50%;
  }

  &::after {
    right: 50%;
  }

  &::before,
  &::after {
    content: '';
    height: 2px;
    width: 0;
    bottom: 1px;
    position: absolute;
    background: #36d7b7;
    transition: 0.2s ease all;
  }
`,E=Object(c["a"])(`label`)`
  top: -20px;
  flex: 1;
  font-size: 10px;
  line-height: 20px;
  text-align: center;
  color: #999;
  position: absolute;
  pointer-events: none;
  transition: 0.2s ease all;
  width: 50px;
`,D={functional:!0,props:{inputs:{type:Object,default:()=>({})},update:{type:Function,required:!0}},render(e,{props:{inputs:t,update:r}}){return e($,[Object.keys(t).map(i=>e(_,[e(q,{attrs:{name:i,type:"margin"===i?"text":"number",value:t[i]},on:{input:r(i)}}),e(M),e(E,{attrs:{htmlFor:i}},[i])]))])}};var L,T;const U=Object(c["a"])(`li`)`
  flex: 0 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 20px;
  border: 0.5px solid #e2e2e2;
  box-shadow: 1px 1px 1px #e2e2e2;
  width: 300px;
  height: 300px;
  margin: 30px;
  font-size: 18px;
  letter-spacing: 1px;
  position: relative;
`,G=Object(c["a"])(`span`)`
  position: absolute;
  top: 20px;
`;let H=(L=a()({props:{spinner:{type:Object,required:!0},name:{type:String,required:!0},color:{type:String,required:!0}}}),L(T=class extends n["default"]{constructor(...e){var t;return t=super(...e),Object(o["a"])(this,"defaults",{}),t}beforeMount(){const e=Object.entries(this.spinner.props).filter(([e,t])=>t.default),t=e.reduce((e,[t,r])=>{return e[t]=r.default,e},{}),r=e.reduce((e,[t,r])=>{return e[t]=r.type,e},{});delete t.color,delete t.loading,delete t.sizeUnit,delete t.widthUnit,delete t.heightUnit,delete t.radiusUnit,delete t.loaderStyle,this.defaults=t,this.propTypes=r}update(e){return t=>{this.defaults[e]=this.propTypes[e](t.target.value)}}renderSpinner(e){const t=this.$createElement;return t(e,{props:Object(z["a"])({color:this.color},this.defaults)})}render(){const e=arguments[0];return e(U,[e(G,[this.name]),this.renderSpinner(this.spinner),e(D,{attrs:{inputs:this.defaults,update:this.update.bind(this)}})])}})||T);var J;r("a35d");const Y=Object(c["a"])(`section`)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-width: 300px;
  width: 80%;
  margin: 0 auto;
`,B=Object(c["a"])(`ul`)`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  background: #fff;
`;let N=a()(J=class extends n["default"]{constructor(...e){var t;return t=super(...e),Object(o["a"])(this,"color","#36D7B7"),t}updateColor(e){this.color=e.hex}renderSpinner(e){const t=this.$createElement;return t(e,{attrs:{color:this.color}})}render(){const e=arguments[0];return e(C,{attrs:{color:this.color}},[e(w,{attrs:{color:this.color,updateColor:this.updateColor}}),e(Y,[e(u),e(B,[Object.entries(l.a).map(([t,r])=>e(H,{attrs:{color:this.color,name:t,spinner:r}}))])])])}})||J;new n["default"]({el:"#app",render:e=>e(N)})},a35d:function(e,t,r){}});
